README.md

```diff
@@ -2,80 +2,203 @@
+Workflow engine using [WDL](https://github.com/broadinstitute/wdl/blob/wdl2/SPEC.md) as the workflow and task language.
 
-API Documenation
-----------------
+<!---toc start-->
+
+* [Cromwell](#cromwell)
+  * [API Documenation](#api-documenation)
+  * [Scala API Usage](#scala-api-usage)
+  * [REST API](#rest-api)
+    * [POST /workflows](#post-workflows)
+    * [GET /workflow/:id/status](#get-workflowidstatus)
+    * [GET /workflow/:id/outputs](#get-workflowidoutputs)
+* [Developer](#developer)
+  * [Generate WDL Parser](#generate-wdl-parser)
+  * [Generating and Hosting ScalaDoc](#generating-and-hosting-scaladoc)
+  * [Architecture](#architecture)
+    * [cromwell.parser](#cromwellparser)
+    * [cromwell.binding](#cromwellbinding)
+    * [cromwell.backend](#cromwellbackend)
+    * [cromwell.engine](#cromwellengine)
+
+<!---toc end-->
+
+## API Documenation
 
 API Documentation can be found [here](http://broadinstitute.github.io/cromwell/scaladoc).
 
-Generate WDL Parser
--------------------
+Note that this may not be completely up to date or even useful at this time.
 
-Install the latest version of [Hermes](http://github.com/scottfrazer/hermes), then run the following command within this directory:
+## Scala API Usage
 
+The main entry point into the parser is the `WdlNamespace` object.  A WDL file is considered a namespace, and other namespaces can be included by using the `import` statement (but only with an `as` clause).
+
+```scala
+import cromwell.binding.WdlNamespace
+
+object Main extends App {
+  val ns = WdlNamespace.load(new File("test.wdl"))
+  val ns2 = WdlNamespace.load("workflow wf {}")
+  val workflows = ns.workflows
+  val tasks = ns.tasks
+  val commands = ns.tasks.map{_.command}
+}
 ```
-hermes generate src/main/resources/grammar.hgr --language=java --directory=src/main/java --name=wdl --java-package=cromwell.parser --java-use-apache-commons
+
+To access only the parser, use the `AstTools` library, as follows:
+
+```scala
+import cromwell.binding.WdlNamespace
+
+object Main extends App {
+  val ast = AstTools.getAst(new File(args(0)))
+
+  // Second parameter is a descriptor about where the first string came from.
+  // Most of the time this would be the URI of where the text was loaded from,
+  // but there are no restrictions on what the string can be.
+  val astFromString = AstTools.getAst("workflow wf {}", "string")
+
+  println(ast.toPrettyString)
+}
 ```
 
-Generating and Hosting ScalaDoc
--------------------------------
+## Command Line Usage
 
-Essentially run `sbt doc` then commit the generated code into the `gh-pages` branch on this repository
+Run the JAR file with no arguments to get the usage message:
 
 ```
-$ sbt doc
-$ git co gh-pages
-$ mv target/scala-2.11/api scaladoc
-$ git add scaladoc
-$ git commit -m "API Docs"
-$ git push origin gh-pages
+$ java -jar target/scala-2.11/cromwell-0.1-SNAPSHOT.jar
+
+java -jar cromwell.jar <action> <parameters>
+
+Actions:
+
+validate <WDL file>
```

Comment:
It's actually completely case insensitive (which I like, but I could see some wacko disliking)